app/renderer/actions/Session.js

```diff
@@ -306,8 +306,11 @@ export function newSession (caps, attachSessId = null) {
         port = session.server.browserstack.port = process.env.BROWSERSTACK_PORT || 443;
         path = session.server.browserstack.path = '/wd/hub';
         username = session.server.browserstack.username || process.env.BROWSERSTACK_USERNAME;
-        desiredCapabilities['bstack:options'] = {};
-        desiredCapabilities['bstack:options'].source = 'appiumdesktop';
+        if (desiredCapabilities['bstack:options']) {
+          desiredCapabilities['bstack:options'].source = 'appiumdesktop';
+        } else {
+          desiredCapabilities.source = 'appiumdesktop';
```

Comment:
so you are intentionally setting a non-w3c style capability to the main level of the capabilities object? why not put it on `bstack:options` to be safe? (and simply create `bstack:options` if it doesn't exist, like the previous code did)

Reply:
The current behaviour was resetting the user's passed bstack:options to a empty object. Due to this all the properties passed in bstack:options during w3c session never reached browserstack.
And at browserstack end we use source to detect any session ( mjsonp or w3c) run using appium desktop 
1. `bstack:options` are passed when it is a w3c session, hence during w3c session , adding source into `bstack:options`
2. In mjsonp sessions, all the w3c, non w3c capabilities are passed directly inside desiredCapabilities, hence added in desiredCapabilities
I hope this answers your question @jlipps

Reply:
i guess i still don't see why you wouldn't just do this:

```js
if (!desiredCapabilities['bstack:options']) {
  desiredCapabilities['bstack:options'] = {};
}
desiredCapabilities['bstack:options'].source = 'appiumdesktop';
```

this also presumably fixes the bug but doesn't fall back to using mjsonwp capabilities. we should really only use w3c caps nowadays.

Reply:
`bstack:options` is only used for w3c sessions
For sessions being run from older appium desktop client, the source property is not being added to MJSONWP session.
With the above suggestion source property gets skipped for MJSONWP session hence can't do it using that way

Let me know if there are any other concerns

Reply:
I don't understand your point. This change is only for new appium inspector clients, which use the W3C strategy to create sessions.

Reply:
yeah understood. We just need the change for W3C and handling for MJSONWP session is not required

Reply:
@jlipps please review